src/main.leo

```diff
@@ -143,15 +155,16 @@ program privx_exchange_beta4.aleo {
         Mapping::set(settings, token_id, new_setting);
     }
 
-    transition claim(public token_id: u64) {
-        return then finalize(self.caller, token_id);
+    transition claim(public token_id: u64, account: privx_account.leo/Account) {
```

Comment:
Claim should be claim_private from Record Account to another Record to preserve privacy. Likewise there should be another parallel transition say, claim_public which accepts addresses and modifies mapping balance